### PR TITLE
Initial Aruba Documentation

### PR DESCRIPTION
**Pull request by qcorporation**
## Change Log
- added manifest file with udp and tcp support
- named ownership for aruba to the dnd team
- documented all fields from the official aruba documentation and mapped them to either ecs or fields
- put placeholders for stream.yml, base-fields and docker-compose
- added example logs from CX 6300, 6000 and 8360

## Checklist
- [ x ] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [ no ] I have verified that all data streams collect metrics or logs.
- [ x ] I have added an entry to my package's `changelog.yml` file.
- [ x ] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist
- [ ] Reviewed that all ecs or newly created maps make sense in the absence of actual logs, just from the documentation
- [ ] Review al fields data types, that they make sense
- [ ] Validate that nothing was missed for message type for the Aruba CX lines between version v5200 -> v8214

## How to test this PR locally
- review the mappings. This will be a guide for additional contributions as we separate the work

### CODE REVIEW

File: packages/hpe_aruba_cx/data_stream/log/fields/fields.yml [hunk 1]
```diff
@@ -0,0 +1,809 @@
...
+          
+        - name: type
+          type: keyword
+          description: >
+          
+    - name: vrf
+      type: group
+      fields:
+        - name: id
+          type: long
```
Comment (gogochan) [inline packages/hpe_aruba_cx/data_stream/log/fields/fields.yml:790]: should this also be `keyword`?

Comment (qcorporation) [reply packages/hpe_aruba_cx/data_stream/log/fields/fields.yml]: I agree, changed to `keyword`

File: sonar-project.properties [hunk 2]
```diff
@@ -7,7 +7,7 @@ sonar.sourceEncoding=UTF-8
 
 # Coverage report created by elastic-package just reference to *.yml files
 # Remove all these extensions to try to keep just *.yml files
-sonar.exclusions=**/*.java,**/Dockerfile,**/*.tf,**/*.go,**/*.yaml,**/*.xml,**/*.md,**/*.jsp
```
Comment (taylor-swanson) [inline sonar-project.properties:10]: We should make this change in a separate PR. Otherwise it drags ecosystem in for this review, which is probably unnecessary.

Comment (qcorporation) [reply sonar-project.properties]: ok - reverting

File: packages/hpe_aruba_cx/changelog.yml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+# newer versions go on top
+- version: "0.1.0"
+  changes:
+    - description: Official support for HPE Aruba CX
+      type: enhancement
+      link: https://github.com/elastic/integrations/pull/1 # FIXME Replace with the real PR link
```
Comment (taylor-swanson) [inline packages/hpe_aruba_cx/changelog.yml:6]: I imagine you'll wait for the final merge to main to update this?

File: packages/hpe_aruba_cx/data_stream/log/agent/stream/udp.yml.hbs [hunk 4]
```diff
@@ -0,0 +1,15 @@
+host: "{{udp_host}}:{{udp_port}}"
```
Comment (taylor-swanson) [inline packages/hpe_aruba_cx/data_stream/log/agent/stream/udp.yml.hbs:1]:
```suggestion
host: "{{listen_address}}:{{listen_port}}"
```

File: packages/hpe_aruba_cx/data_stream/log/manifest.yml [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+        show_user: false
+      - name: internal_networks
+        type: text
+        title: Internal Networks
+        description: List of internal networks. Supports IPv4 and IPv6 addresses and ranges in CIDR notation. Also supports the named ranges listed [here](https://www.elastic.co/guide/en/elasticsearch/reference/current/network-direction-processor.html#supported-named-network-ranges).
+        multi: true
+        required: false
+        show_user: false
+        default:
+          - private
```
Comment (taylor-swanson) [inline packages/hpe_aruba_cx/data_stream/log/manifest.yml:54]: Are we intending to use these values? I don't see them referenced in the agent handle bars configs.

File: packages/hpe_aruba_cx/_dev/deploy/docker/docker-compose.yml [hunk 6]
```diff
@@ -0,0 +1,7 @@
+version: '2.3'
```
Comment (taylor-swanson) [inline packages/hpe_aruba_cx/_dev/deploy/docker/docker-compose.yml:1]: Is this just a stub for now? We could add UDP (and a log one if you add filestream) as well here.

File: packages/hpe_aruba_cx/data_stream/log/elasticsearch/ingest_pipeline/default.yml [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+  - set:
+      field: ecs.version
+      value: '8.11.0'
+
+  #
+  # Parse the syslog message
+  # TBD
+  #
+
+
```
Comment (taylor-swanson) [inline packages/hpe_aruba_cx/data_stream/log/elasticsearch/ingest_pipeline/default.yml:18]: We'll want an `on_failure` handler, but we can handle that in the next part.

**Review by taylor-swanson — COMMENTED** [review]
I don't see a filestream input, we should probably add one. Even if the product itself doesn't export directly to a log, we usually add one as the user may have an environment where the logs eventually get to us in the form of a file.

File: packages/hpe_aruba_cx/data_stream/log/fields/fields.yml [hunk 8]
```diff
@@ -0,0 +1,809 @@
+- name: aruba
+  type: group
+  fields:
```
Comment (dwhyrock) [inline packages/hpe_aruba_cx/data_stream/log/fields/fields.yml:3]: I'm curious how you populated this file.  Did you copy another integration's `fields.yml` and remove the descriptions?  Or did you go into the logs and derive which fields we would need to populate?

File: packages/hpe_aruba_cx/data_stream/log/manifest.yml [hunk 9]
```diff
@@ -0,0 +1,169 @@
...
+        show_user: true
+        default: localhost
+      - name: listen_port
+        type: integer
+        title: Listen Port
+        description: The UDP port number to listen on default is 514.
+        multi: false
+        required: true
+        show_user: true
+        default: 514
```
[the remark this answers is not in the document]
Comment (dwhyrock) [reply packages/hpe_aruba_cx/data_stream/log/manifest.yml:132]: good call on not using < 1024.

File: packages/hpe_aruba_cx/docs/README.md [hunk 10]
```diff
@@ -0,0 +1,1453 @@
...
+## Logs
+### Exported fields
+
+Below are the fields from the different event types and their mapping into ECS supported fields or customer Aruba fields
+
+To Be Removed
+Note: Field types are defined within `fields.yml`
+Note: Descriptions have not been filled out
+
+#### [AAA events](https://www.arubanetworks.com/techdocs/AOS-CX/10.07/HTML/5200-8214/Content/events/AAA.htm)
```
Comment (dwhyrock) [inline packages/hpe_aruba_cx/docs/README.md:45]: Also curious about the workflow here.  Was this automated in some way?  Or was this as tedious as I am imagining it would be?

**Review by dwhyrock — COMMENTED** [review]
I mostly have some questions about how this was done, especially about all the fields and how they were created/generated.

Overall it looks like a solid start to the integration.  

I'll withhold my approval since it looks like CI isn't quite passing:

```
-| aruba.vrf.id |  | long |
+| aruba.vrf.id |  | keyword |
 | aruba.vrf.name |  | keyword |
Error: checking package failed: checking readme files are up-to-date failed: files do not match
```

https://buildkite.com/elastic/integrations/builds/16001#0191fffc-07da-4036-903d-c3e19b432020/445-451

File: packages/hpe_aruba_cx/data_stream/log/manifest.yml (hunk 9 above)
[the remark this answers is not in the document]
Comment (qcorporation) [reply packages/hpe_aruba_cx/data_stream/log/manifest.yml:132]: If you take a look, the Aruba instructions at the CLI level for setting up log forward: https://www.arubanetworks.com/techdocs/AOS-CX/AOSCX-CLI-Bank/cli_4100i/Content/Chp_RSyslog/RSyslog_cmds/log-10.htm

The defaults are as follows
```
udp [<PORT-NUM>] | Range: 1 to 65535. Default: 514
tcp [<PORT-NUM>] | Range: 1 to 65535. Default: 1470
tls [<PORT-NUM>] | Range: 1 to 65535. Default: 6514
```

Simply trying to adhere to the same defaults from the Aruba documentation

File: packages/hpe_aruba_cx/changelog.yml (hunk 3 above)
Comment (qcorporation) [reply packages/hpe_aruba_cx/changelog.yml:6]: yes, that'd be the final changes when we create the final PR into `main`

File: packages/hpe_aruba_cx/_dev/deploy/docker/docker-compose.yml (hunk 6 above)
Comment (qcorporation) [reply packages/hpe_aruba_cx/_dev/deploy/docker/docker-compose.yml:1]: It's a stub for now - WIP for the UDP side of things.

For filestream, I did an initial look at the setup for Aruba CX through their documentation and I didn't see a way to ingest logs via file

File: packages/hpe_aruba_cx/data_stream/log/agent/stream/udp.yml.hbs (hunk 4 above)
Comment (qcorporation) [reply packages/hpe_aruba_cx/data_stream/log/agent/stream/udp.yml.hbs]: fixed.

File: packages/hpe_aruba_cx/data_stream/log/manifest.yml (hunk 5 above)
Comment (qcorporation) [reply packages/hpe_aruba_cx/data_stream/log/manifest.yml]: removed from source

File: packages/hpe_aruba_cx/_dev/deploy/docker/docker-compose.yml (hunk 6 above)
Comment (qcorporation) [reply packages/hpe_aruba_cx/_dev/deploy/docker/docker-compose.yml:1]: added udp, tls, filestream within docker-compose file

File: packages/hpe_aruba_cx/data_stream/log/elasticsearch/ingest_pipeline/default.yml (hunk 7 above)
Comment (qcorporation) [reply packages/hpe_aruba_cx/data_stream/log/elasticsearch/ingest_pipeline/default.yml]: added `on_failure` catch-call

File: packages/hpe_aruba_cx/_dev/build/build.yml [hunk 11]
```diff
@@ -0,0 +1,3 @@
+dependencies:
+  ecs:
+    reference: "git@v8.11.0"
```
[the remark this answers is not in the document]
Comment (qcorporation) [reply packages/hpe_aruba_cx/_dev/build/build.yml:3]: @dwhyrock good question - it looks like we point to the latest ECS release, which at this point is 8.11.0
In general, our integrations for the foreseeable future will always speak ECS.

In contrast, if we were to pick an older ECS version, then we would not be able to specify fields that are released in a later release. That will limit our 'language' in which we can describe the log message that is coming from the appliance.

File: packages/hpe_aruba_cx/data_stream/log/fields/fields.yml (hunk 8 above)
Comment (qcorporation) [reply packages/hpe_aruba_cx/data_stream/log/fields/fields.yml:3]: see this [response](https://github.com/elastic/integrations/pull/11101#discussion_r1767560652) for more details

For this file, I auto-generated using prompting into GitHub Copilot after I got the maps down in a file
